donkeycar/contrib/gradcam/cli.py

```diff
@@ -0,0 +1,432 @@
+                catalog = json.load(f)
+
+            for record in catalog:
+                if 'image_path' in record:
+                    img_path = os.path.join(data_path, record['image_path'])
+                    metadata[img_path] = {
+                        'steering': record.get('steering'),
+                        'throttle': record.get('throttle')
+                    }
+            print(f"Loaded metadata for {len(metadata)} images from catalog")
+        except Exception as e:
+            print(f"Error loading catalog.json: {e}")
+
+    # If no catalog or few entries, look for individual JSON files
+    if len(metadata) == 0:
+        print("Looking for individual JSON files")
+        json_files = glob.glob(os.path.join(data_path, "**/*.json"),
+                               recursive=True)
+
+        for json_path in json_files:
+            try:
+                img_path = os.path.splitext(json_path)[0] + '.jpg'
+                if os.path.exists(img_path):
+                    with open(json_path, 'r') as f:
+                        metadata[img_path] = json.load(f)
+            except Exception as e:
+                print(f"Error reading {json_path}: {e}")
+
+        print(f"Found {len(metadata)} JSON files with metadata")
+
+    return metadata
+
+
+def process_images(image_paths, model, last_conv_layer_name, metadata={},
+                   stride=1, max_frames=None, alpha=0.4, exclude_regions=None):
+    """
+    Process images and generate Grad-CAM visualizations.
+
+    Args:
+        image_paths: List of image file paths
+        model: Loaded Keras model
+        last_conv_layer_name: Name of the last convolutional layer
+        metadata: Dictionary of image metadata
+        stride: Process every Nth frame
+        max_frames: Maximum number of frames to process
+        alpha: Transparency of the heatmap overlay
+        exclude_regions: Regions to exclude from the heatmap
+
+    Returns:
+        List of visualization frames
+    """
+    # Apply stride and limit
+    image_paths = image_paths[::stride]
+    if max_frames is not None and max_frames > 0:
+        image_paths = image_paths[:max_frames]
+
+    print(f"Processing {len(image_paths)} images")
+
+    # Sort images numerically if they follow a numeric naming pattern
+    try:
+        def extract_number(filename):
+            base = os.path.basename(filename)
+            parts = base.split('_')
+            if parts and parts[0].isdigit():
+                return int(parts[0])
+            return filename
+
+        image_paths = sorted(image_paths, key=extract_number)
+        numbers = [extract_number(path) for path in image_paths]
+        print(f"Frame sequence: first={numbers[0]}, last={numbers[-1]}")
+        print(f"Sample sequence: {numbers[:10]} ...")
+    except Exception as e:
+        print(f"Could not sort images numerically: {e}")
+
+    # Process images
+    frames = []
+    for i, img_path in enumerate(image_paths):
```

Comment:
The code inside this loop is the core part of the PR. This is the bit that should be translated into `MakeMovie` such that we can choose between the current and your new version of displaying the salient features.  At that point all the tub data loading and file loading has already happened. This will greatly simplify your PR.